From d6ff5dc9e9829b437dead4ae563281dc4635cf48 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Horia=20Geant=C4=83?= <horia.geanta@nxp.com>
Date: Fri, 10 Jun 2016 15:58:45 +0300
Subject: [PATCH 285/508] crypto: caam/qi - register algos only when QI is
 present
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Do not rely solely on the build system, since all armv8 SoCs use
the same defconfig (and thus CONFIG_FSL_QMAN=y even for platforms
like LS1012A). Instead, double-check QI presence as reported by HW.

Signed-off-by: Horia GeantÄƒ <horia.geanta@nxp.com>
[Original patch taken from SDK-V2.0-1703]
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/crypto/caam/caamalg_qi.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/crypto/caam/caamalg_qi.c b/drivers/crypto/caam/caamalg_qi.c
index 7ee3103..7e88a2e 100644
--- a/drivers/crypto/caam/caamalg_qi.c
+++ b/drivers/crypto/caam/caamalg_qi.c
@@ -2265,7 +2265,7 @@ static int __init caam_qi_algapi_init(void)
 	* If priv is NULL, it's probably because the caam driver wasn't
 	* properly initialized (e.g. RNG4 init failed). Thus, bail out here.
 	*/
-	if (!priv)
+	if (!priv || !priv->qi_present)
 		return -ENODEV;
 
 	INIT_LIST_HEAD(&alg_list);
-- 
1.7.5.4

